seqr/views/utils/dataset_utils.py

```diff
@@ -449,3 +450,115 @@ def _load_rna_seq(model_cls, file_path, user, mapping_file, ignore_extra_samples
         logger.warning(warning, user)
 
     return samples_to_load, info, warnings
+
+
+TOOL_FIELD = 'tool'
+PHENOTYPE_PRI_HEADER = {
+    'tool': TOOL_FIELD, 'project': 'project', 'sampleId': 'sample_id', 'rank': 'rank', 'geneId': 'gene_id',
+    'diseaseId': 'disease_id', 'diseaseName': 'disease_name', 'scoreName1': 'score_name1', 'score1': 'score1',
+    'scoreName2': 'score_name2', 'score2': 'score2', 'scoreName3': 'score_name3', 'score3': 'score3'}
+
+
+def _parse_phenotype_pri_row(row):
+    record = {mapped_key: row[key] for key, mapped_key in PHENOTYPE_PRI_HEADER.items()}
+
+    tool = PhenotypePrioritization.TOOL_LOOKUP.get(record[TOOL_FIELD], None)
+    if not tool:
+        raise ValueError('Expecting {} for the "tool" column but found {}'.format(
+            ', '.join([v for k, v in PhenotypePrioritization.TOOL_CHOICES]), record[TOOL_FIELD]))
+    record[TOOL_FIELD] = tool
+
+    scores = {record.pop('score_name1'): record.pop('score1')}
+    for score_index in ['2', '3']:
+        score_name = record.pop('score_name' + score_index, None)
+        score = record.pop('score' + score_index, None)
+        if score_name:
+            scores[score_name] = score
+    record['scores'] = scores
+
+    return record
+
+
+def _load_phenotype_pri_file(file_path):
+    data_by_id = defaultdict(dict)
+    f = file_iter(file_path)
+    header = _parse_tsv_row(next(f))
+    missing_cols = [col for col in PHENOTYPE_PRI_HEADER.keys() if col not in header]
+    if missing_cols:
+        raise ValueError(f'Invalid file: missing column(s) {", ".join(missing_cols)}')
+
+    count = 0
+    for line in tqdm(f, unit=' rows'):
+        row = dict(zip(header, _parse_tsv_row(line)))
+        record = _parse_phenotype_pri_row(row)
+        sample_id = record.pop('sample_id', None)
+        project = record.pop('project', None)
+        if not sample_id or not project:
+            raise ValueError('Both sample ID and project fields are required.')
+        if data_by_id[sample_id]:
+            if project != data_by_id[sample_id]['project']:  # a sample must belong to a single project
+                raise ValueError(f'Invalid project name for sample {sample_id}')
+            data_by_id[sample_id]['records'].append(record)
+        else:
+            data_by_id[sample_id]['project'] = project
+            data_by_id[sample_id]['records'] = [record]
+        count += 1
+
+    return count, data_by_id
+
+
+def load_phenotype_pri(file_path, user, ignore_extra_samples):
+    count, data_by_id = _load_phenotype_pri_file(file_path)
+
+    message = f'Parsed {count} LIRICAL/Exomiser data records in {len(data_by_id)} samples'
+    info = [message]
+    logger.info(message, user)
+
+    indivs = Individual.objects.filter(individual_id__in=data_by_id.keys())
+    prefetch_related_objects(indivs, 'family__project')
+    existing_indivs_by_id = {ind.individual_id: ind for ind in indivs
```

Comment:
`individual_id` is not unique across projects, and I think the code in this file could have serious unintended consequences for data in unrelated projects because of that. While it is slightly less performant, I think it is safer to do this all on a per-project basis, and if multiple projects are represented in a file you would do the updates multiple times:
```
individual_records = {}
for project, sample_records in data_by_id.items():
    indivs = Individual.objects.filter(
        project__name=project, individual_id__in=sample_records.keys()).prefetch_related('family__project')
    existing_indivs_by_id = {ind.individual_id: ind for ind in invids}
    ...
   individual_records[existing_indivs_by_id[sample_id]] = sample_records[sample_id]
```